src/panels/config/scene/ha-scene-editor.ts

```diff
@@ -0,0 +1,727 @@
+    this._entities = [...this._entities, ...deviceEntities];
+    deviceEntities.forEach((entity) => {
+      this._storeState(entity);
+    });
+    this._dirty = true;
+  }
+
+  private _deleteDevice(ev: Event) {
+    const deviceId = (ev.target as any).device;
+    this._devices = this._devices.filter((device) => device !== deviceId);
+    const deviceEntities = this._deviceEntityLookup[deviceId];
+    this._entities = this._entities.filter(
+      (entity) => !deviceEntities.includes(entity)
+    );
+  }
+
+  private _nameChanged(ev: CustomEvent) {
+    if (!this._config || this._config.name === ev.detail.value) {
+      return;
+    }
+    this._config.name = ev.detail.value;
+    this._dirty = true;
+  }
+
+  private _stateChanged(event: HassEvent) {
+    if (this._entities.includes(event.data.entity_id)) {
+      this._dirty = true;
+    }
+  }
+
+  private _backTapped(): void {
+    if (this._dirty) {
+      showConfirmationDialog(this, {
+        text: this.hass!.localize(
+          "ui.panel.config.scene.editor.unsaved_confirm"
+        ),
+        confirmBtnText: this.hass!.localize("ui.common.yes"),
+        cancelBtnText: this.hass!.localize("ui.common.no"),
+        confirm: () => this._goBack(),
+      });
+    } else {
+      this._goBack();
+    }
+  }
+
+  private _goBack(): void {
+    applyScene(this.hass, this._storedStates);
+    history.back();
+  }
+
+  private _deleteTapped(): void {
+    showConfirmationDialog(this, {
+      text: this.hass!.localize("ui.panel.config.scene.picker.delete_confirm"),
+      confirmBtnText: this.hass!.localize("ui.common.yes"),
+      cancelBtnText: this.hass!.localize("ui.common.no"),
+      confirm: () => this._delete(),
+    });
+  }
+
+  private async _delete(): Promise<void> {
+    await deleteScene(this.hass, this.scene!.attributes.id!);
+    applyScene(this.hass, this._storedStates);
+    history.back();
+  }
+
+  private _calculateStates(): EntityStates {
+    const output: EntityStates = {};
+    this._entities.forEach((entity) => {
+      const state = this._getCurrentState(entity);
+      if (state) {
+        output[entity] = state;
+      }
+    });
+    return output;
+  }
+
+  private _storeState(entity: string): void {
+    if (entity in this._storedStates) {
+      return;
+    }
+    const state = this._getCurrentState(entity);
+    if (!state) {
+      return;
+    }
+    this._storedStates[entity] = state;
+  }
+
+  private _getCurrentState(entity: string) {
+    const stateObj = this.hass.states[entity];
+    if (!stateObj) {
+      return;
+    }
+    const domain = computeDomain(entity);
+    const attributes = {};
+    for (const attribute in stateObj.attributes) {
+      if (
+        SAVED_ATTRIBUTES[domain] &&
+        SAVED_ATTRIBUTES[domain].includes(attribute)
+      ) {
+        attributes[attribute] = stateObj.attributes[attribute];
+      }
+    }
+    return { state: stateObj.state, ...attributes };
```

Comment:
turn this around. We wouldn't want `state` inside attributes to override the actual state.